### PR TITLE
[ WIP ] Added new implementation of loop-protect  (#698)

### PR DESCRIPTION
**Pull request by siddhant1**
I have verified that this pull request:

* [x ] has no linting errors (`npm run lint`)
* [x ] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [ x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

**Comment by siddhant1** [comment]
This code is successfully catching for(var) , while and do while infinite loops but stucks on for(let) loop. I have raised an issue , @catarak any possible implementation that you can think of rn?

**Comment by siddhant1** [comment]
@catarak , I got a lead for his one , codesandbox is implementing loop-protection , I will have a look at their code and make progress with this one.
Meanwhile , have you got any leads , I might need some help with this one

File: client/utils/loopProtect.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+const generateBefore = (t, id) =>
```
Comment (catarak) [inline client/utils/loopProtect.js]: where did you pull this file from?

**Comment by catarak** [comment]
just took a look at this... you are definitely on the right track! are you pulling in `loop-protect` from npm or github? it seems like you are not, and i think that's the right approach.

**Pull request closed without merging**